src/prefect/cli/deploy.py

```diff
@@ -433,6 +435,51 @@ async def _run_single_deploy(
             console=app.console, client=client
         )
 
+    docker_build_steps = [
+        "prefect_docker.deployments.steps.build_docker_image",
+        "prefect_docker.projects.steps.build_docker_image",
+    ]
+
+    docker_build_step_exists = any(
+        any(step in action for step in docker_build_steps)
+        for action in deploy_config.get("build", actions.get("build")) or []
+    )
+
+    if is_interactive() and not docker_build_step_exists:
+        work_pool = await client.read_work_pool(deploy_config["work_pool"]["name"])
+        docker_based_infrastructure = "image" in work_pool.base_job_template.get(
+            "variables", {}
+        ).get("properties", {})
+        if docker_based_infrastructure:
+            build_docker_image_step = await prompt_build_custom_docker_image(
+                app.console, deploy_config
+            )
+            if build_docker_image_step is not None:
+                if actions.get("build"):
+                    actions["build"].append(build_docker_image_step)
+                else:
+                    actions["build"] = [build_docker_image_step]
+
+                work_pool_job_variables_image_not_found = not get_from_dict(
+                    deploy_config, "work_pool.job_variables.image"
+                )
+                if work_pool_job_variables_image_not_found:
+                    deploy_config["work_pool"]["job_variables"][
+                        "image"
+                    ] = "{{ build-image.image }}"
```

Comment:
If not found, I should also update the `work_pool` itself so it reflects in the UI.